src/bokeh/models/nodes.py

```diff
@@ -0,0 +1,214 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2012 - 2023, Anaconda, Inc., and Bokeh Contributors.
+# All rights reserved.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+#-----------------------------------------------------------------------------
+# Boilerplate
+#-----------------------------------------------------------------------------
+from __future__ import annotations
+
+import logging # isort:skip
+log = logging.getLogger(__name__)
+
+#-----------------------------------------------------------------------------
+# Imports
+#-----------------------------------------------------------------------------
+
+# Standard library imports
+from typing import Any, Literal
+
+# Bokeh imports
+from ..core.properties import (
+    Either,
+    Enum,
+    Instance,
+    Int,
+    Required,
+    String,
+)
+from ..core.property.aliases import CoordinateLike
+from ..model import Model
+
+#-----------------------------------------------------------------------------
+# Globals and constants
+#-----------------------------------------------------------------------------
+
+__all__ = (
+    "Indexed",
+    "Node",
+    "XY",
+)
+
+#-----------------------------------------------------------------------------
+# General API
+#-----------------------------------------------------------------------------
+
+ImplicitTarget = Literal["viewport", "canvas", "plot", "frame", "parent"]
```

Comment:
"Implicit" does not seem very descriptive here, since this is used to explicitly name some target or other. `BuiltInTarget` maybe?

Reply:
There are two kinds of symbolic nodes: with explicit (model reference) and implicit (named) targets. The implicit case is a hack for the most part and an effect of not having explicit access to plot's frame and/or canvas. If e.g. PR #13286 was completed earlier, then this wouldn't be necessary, except perhaps for `"parent"` case and maybe `"viewport"`. Then I would remove this enum and add models for those cases.

I don't mind changing this to `BuiltinTarget`, but at least from my perspective, `ImplicitTarget` conveys that fact that from the perspective of a model that uses an implicit node, it's not clear what it refers to, which canvas, plot or frame it refers to. However, as I mentioned in a meeting a few weeks back, the node API is the least refined and most experimental change in this PR, and I expect substantial changes in the near future.